app/controllers/support/devices/order_status_controller.rb

```diff
@@ -0,0 +1,35 @@
+class Support::Devices::OrderStatusController < Support::BaseController
+  before_action :set_school
+
+  def edit
+    @form = Support::EnableOrdersForm.new(enable_orders_form_params)
+  end
+
+  def update
+    @form = Support::EnableOrdersForm.new(enable_orders_form_params)
+    if @form.valid?
+      ActiveRecord::Base.transaction do
+        allocation = SchoolDeviceAllocation.find_or_initialize_by(school_id: @school.id, device_type: 'std_device')
+        # we only take the cap from the user if they chose specific circumstances
+        # for both other states, we need to infer a new cap from the chosen state
+        allocation.cap = allocation.cap_implied_by_order_state(order_state: @form.order_state, given_cap: @form.cap)
+        allocation.save!
+        @school.update!(order_state: @form.order_state)
```

Comment:
It feels like the controller is doing too much here. Could we extract this logic into something else (service? model?) in subsequent PRs?

Reply:
I agree, and I was thinking that at the time I submitted this, but I wanted to get the Computacenter API call in there too and then see if that introduced any further complications.

That's now done (in #468) and I think there's a clear chunk of code I can pull out to a separate ... _something_  - I'll do that in a separate PR